vm/src/exceptions.rs

```diff
@@ -859,6 +861,24 @@ fn key_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
     }
 }
 
+fn os_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
+    let args = exc.args();
+    if args.as_slice().len() == 2 {
+        // SAFETY: len() == 2 is checked so get_arg 1 or 2 won't panic
+        let s = match (
+            exc.get_arg(0).unwrap().downcast_exact::<PyInt>(vm),
+            exc.get_arg(1).unwrap().downcast_exact::<PyStr>(vm),
+        ) {
+            (Ok(errno), Ok(msg)) => format!("[Errno {}] {}", errno, msg),
+            (Ok(errno), Err(_)) => format!("[Errno {}]", errno),
+            (Err(_), _) => "Error".to_owned(),
+        };
```

Comment:
Seems like CPython just calls `__str__` on whatever is in these args and formats them:

```python
>>> class Foo:
...     def __str__(self):
...             return "__str__"
...     def __repr__(self):
...             return "__repr__"
... 
>>> OSError(Foo(), Foo()).__str__()
'[Errno __str__] __str__'
```

so this seems like it could be simplified to directly calling `str` whatever the args are as long as two of them exist.